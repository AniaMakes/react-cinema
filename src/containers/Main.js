import React from 'react';
import Search from '../components/Search';
import Results from '../components/Results';

class Main extends React.Component{
	constructor(props) {
		super(props);
		this.state = {
			data: [],
			resultsDisplayed: false,
		}

		this.getData = this.getData.bind(this);
	}

	getData(searchQuery) {
		console.log(searchQuery);
		console.log(typeof(searchQuery));
		fetch(`http://www.omdbapi.com/?apikey=674ff9&s=${searchQuery}`)
			.then(results => results.json())
			.then(data => {
				this.setState({
					data: data.Search
				})
				console.log(this.state.data);
			})
			.catch(err => console.log(err))
	}

	render(){
		return(
			<div className="main-content">
				<Search receiver={this.getData}/>
				<Results results={this.state.data}/>
			</div>
		)
	}
}

export default Main;