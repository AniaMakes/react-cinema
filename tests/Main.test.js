import React from "react";
import Main from "../src/containers/Main"
import {shallow} from "enzyme";
import fetchMock from 'fetch-mock';
import renderer from "react-test-renderer";

describe("Main", () =>{
	it("Main is rendered", ()=>{
		const mock = renderer.create(<Main />).toJSON();
		expect(mock).toMatchSnapshot();
	})

	describe("fetch and display", () => {
		beforeEach(() => {
				fetchMock.get("*", {Search:["alpha", "bravo", "charlie"]})
		})

		it("fetches and displays", done => {
				const wrapper = shallow(<Main />);

				wrapper.instance().getData()

				setTimeout(()=>{
					const updated = wrapper.update();
					const mainComponent = updated.find("Results");

					expect(mainComponent.get(0).props.results).toEqual(["alpha", "bravo", "charlie"]);

				done();
				}, 0);

				afterEach(()=> {
					fetchMock.restore();
			})
		})
	})
})
