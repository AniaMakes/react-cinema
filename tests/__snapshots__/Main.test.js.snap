// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Main is rendered 1`] = `
<div
  className="main-content"
>
  <section
    className="search-block"
  >
    <form
      className="search-form"
      onSubmit={[Function]}
    >
      <label
        className="search-label"
        htmlFor="search-input"
      >
        Search By Name
      </label>
      <div
        className="search-bar"
      >
        <input
          className="search-input"
          id="search-input"
          name="search-input"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          className="submit-btn"
          type="submit"
        >
          Find
        </button>
      </div>
    </form>
  </section>
  <section
    className="results"
  >
    <h2>
      Results
    </h2>
  </section>
</div>
`;
